src/include/daos_srv/pool.h

```diff
@@ -84,6 +85,28 @@ struct ds_pool {
 	void			*sp_metrics[DAOS_NR_MODULE];
 };
 
+/* Pool service crt-event-handling state */
+struct pool_svc_events {
+	ABT_mutex		pse_mutex;
+	ABT_cond		pse_cv;
+	d_list_t		pse_queue;
+	ABT_thread		pse_handler;
+	bool			pse_stop;
+};
+
+/* Pool service */
+struct pool_svc {
```

Comment:
Without a significantly strong reason, I don't think we should just expose the internal pool_svc type. Could we revert these pool_svc changes and do the pool map manipulations in some new PS RPC(s) instead?

Reply:
I have considered the pool_svc usage cases for a while, but did not found some suitable solution. One of the important reasons for exporting such structure is that all RDB related APIs need the "pool_svc" parameter to start the TX, lock, and so on. If we do not export "pool_svc", then related CHK logic have to be moved into pool/rsvc module(s) or introduce some complex (but non-general) callback, that will make related logic to be more dirty than current implementation. I do not think it is expected.

The usage "pool_svc" is not only in this patch but also in subsequent patches. For example, on the engine of PS leader, the CHK logic will do the following:
1. Check whether current engine is the PS leader or not. If yes, then
2. Parse all related pool shards reported information one by one, compare with its local pool map, and then do related reparation, such as refresh the pool map or destroy related pool shard, that may involve in the interaction with admin via CHK leader.
3. Compare pool label from MBS with its local property, and repair if necessary.

In above example, the "pool_svc" is generated in the 1st step, then repeatedly used in the 2nd step, and then the 3rd step. If we do NOT export "pool_svc" in the 1st step, then means that both 2nd and 3rd steps will be wrapped into pool/rsvc, but they are different than other general pool/rsvc logic, looks very dirty and strange.